tf2onnx/optimizer/transpose_optimizer.py

```diff
@@ -101,8 +101,8 @@ def _calculate_new_shape(graph, op):
                 input_shape = self._g.get_shape(op.input[0])
                 if not input_shape:
                     continue
-
-                if (is_nchw_transpose(op) and (input_shape[3] == 1 or (input_shape[1:3] == [1, 1]))) \
+                if (is_nchw_transpose(op) and len(input_shape) == 4 and

```

Comment:
Doesn't `is_nchw_transpose` already ensure the length of the input shape is 4?

Reply:
You are right it does

Reply:
I take that back.  It seems like the whole issue with #1072 is that somehow a transpose node has perm and shape properties with different dimensions.

Reply:
That somehow was the bug i saw, but when reading the code the check seemed redundant, i put it there again.